torch/distributed/algorithms/_checkpoint/checkpoint_wrapper.py

```diff
@@ -148,16 +175,20 @@ def checkpoint_wrapper(
         module (nn.Module):
             The module to be wrapped
         checkpoint_impl (Optional[CheckpointImpl]):
-            The checkpointing implementation to use. Currently only
-            CheckpointImpl.REENTRANT is supported. Note that this will only
+            The checkpointing implementation to use. Note that this will only
             be passed into the ``torch.utils.checkpoint.checkpoint``
             implementation, and is ignored if a custom ``checkpoint_fn`` is
-            specified.
+            specified. Note that for implementations using reentrant checkpoint
+            from ``torch.utils.checkpoint``, keyword arguments will only be
+            supported if ``checkpoint_impl`` is passed as ``CheckpointImpl.REENTRANT`.
         offload_to_cpu (Optional[bool]):
             Whether to offload activations of this wrapped module to CPU. Note
             that if this is specified, ``checkpoint_impl`` and ``checkpoint_fn``
             arguments will be ignored in favor of the activations being
-            offloaded to CPU. Default is ``False``.
+            offloaded to CPU. Default is ``False``. Wrappers with activation
+            offload can be composed with ones that do recomputation-based
+            checkpoint to trade off increased compute versus CPU memory usage
```

Comment:
> trade off increased compute versus CPU memory usage and H2D transfer.

Clarification:
- Activation checkpointing means _increased_ compute.
- Activation offload means _increased_ CPU memory usage and _additional_ H2D transfers.

If this is correct, then can we consider adding:
```suggestion
            checkpoint to trade off increased compute versus increased CPU memory usage
            and additional H2D transfers.
```
(just to be explicit on the directionality of effects in the tradeoff)